src/Backends/InterfaceSettings.vala

```diff
@@ -57,5 +78,44 @@ public class SettingsDaemon.Backends.InterfaceSettings : GLib.Object {
         accounts_service.cursor_size = interface_settings.get_int (CURSOR_SIZE);
         accounts_service.locate_pointer = interface_settings.get_boolean (LOCATE_POINTER);
         accounts_service.text_scaling_factor = interface_settings.get_double (TEXT_SCALING_FACTOR);
+
+        accounts_service.picture_options = background_settings.get_enum (PICTURE_OPTIONS);
+        accounts_service.primary_color = background_settings.get_string (PRIMARY_COLOR);
+    }
+
+    private void sync_background_to_greeter () {
+        // File.new_for_uri creates file with broken get_basename method, so do this
+        var wallpaper_uri = background_settings.get_string (PICTURE_URI);
+        var wallpaper_path = File.new_for_uri (wallpaper_uri).get_path ();
+        var source = File.new_for_path (wallpaper_path);
+
+        var greeter_data_dir = Path.build_filename (Environment.get_variable ("XDG_GREETER_DATA_DIR"), "wallpaper");
+        if (greeter_data_dir == "") {
+            greeter_data_dir = Path.build_filename ("/var/lib/lightdm-data/", Environment.get_user_name (), "wallpaper");
+        }
+        var folder = File.new_for_path (greeter_data_dir);
+        var dest = File.new_for_path (Path.build_filename (greeter_data_dir, source.get_basename ()));
+
+        try {
+            if (folder.query_exists ()) {
+                var enumerator = folder.enumerate_children ("standard::*", FileQueryInfoFlags.NOFOLLOW_SYMLINKS);
+                FileInfo? info = null;
+                while ((info = enumerator.next_file ()) != null) {
+                    enumerator.get_child (info).@delete ();
+                }
+            } else {
+                folder.make_directory_with_parents ();
+            }
+
+            source.copy (dest, FileCopyFlags.OVERWRITE | FileCopyFlags.ALL_METADATA);
```

Comment:
does we need all this? wouldn't `$XDG_GREETER_DATA_DIR/wallpaper.$ext` and `$XDG_GREETER_DATA_DIR/wallpaper-dark.$ext` be enough? this way we can skip this enumerator loop and just do:

```vala
    if (!folder.query_exists ()) { // folder here point to $XDG_GREETER_DATA_DIR
        folder.make_directory_with_parents ();
    }

    source.copy (folder.get_child (wallpaper_name), OVERWRITE | ALL_METADATA); // wallpaper_name would be "wallpaper.png" for a png file, or "wallpaper.xml" for a dynamic one, etc...
```

Reply:
I don't see a straightforward way of getting a file extension. Can we just it and use `$XDG_GREETER_DATA_DIR/wallpaper`?

Reply:
i think this need a test with a animated/dynamic wallpaper, if they end working correct in the greeter, i think it would be better.

Reply:
Just tested it, Greeter doesn't support dynamic wallpapers, instead it falls back to `/usr/share/backgrounds/elementaryos-default` file.

Reply:
Oh just got that you're talking about file without extensions sorry

What if we just check it's mimetype?

Reply:
Tested again, now with `wallpaper` file, greeter falls back in this case too.

Reply:
I think Gdk.Pixbuf will already check it for us, the problematic case was dynamic ones, since theses are XML files, and sometimes they don't have the XML header that is used to define the mimetype when the file has no extension.

but since we don't support them in the greeter, we don't need to worry about that.

Reply:
> Tested again, now with `wallpaper` file, greeter falls back in this case too.

wait, you tested a static or dynamic one? if it a static one, and fails, the fallback one would fail too.

Reply:
I tested both. 

> if it a static one, and fails, the fallback one would fail too.

Why?

Reply:
because it doesn't have a extension too, so if Gdk.Pixbuf fails to check the mimetype for the `wallpaper` named one it would fail in the `elementaryos-default` too.

Reply:
Tested here, and `\var\lib\lightdm-data\$user\wallpaper` woked fine with my current wallpaper (a .jpg file).